src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -718,7 +707,7 @@ void ov::npuw::JustInferRequest::function_prologue(std::size_t idx) {
     // 2. Unpack the function closure -- right here, if pipelining if not enabled.
     // If it is enabled, the flow is a little bit different - see run_subrequest_for_success()
     // for details.
-    if (!is_pipelined(idx)) {
+    if (!is_pipelined(idx) && m_closure_update_required && !func_desc.forced_to_fcall) {
```

Comment:
Wait... `is_pipelined` checks `&& !desc.forced_to_fcall` - how does it work with `&& !func_desc.forced_to_fcall`?

Reply:
it works exactly as written:
- `is_pipelined` returns `true` for any functions with more than 1 instance. So we don't pipeline FCFAlled functions now. But it also takes the `m_use_function_pipelining` into account.
- the new assumption here is that we don't need to unpack closures for the FCFAlled functions - as we do it just once in the constructor.

So, as a negation of this rule, we do unpack_closure() right here right now if:
1. this function is NOT pipelined (otherwise we do it asynchronously)
2. closure_update_requried (FOLDING) and this function is not FCFAlled.